### PR TITLE
feat(icon): use icon type names

### PR DESCRIPTION
**Pull request by chrisadubois**
# IMPORTANT

**Currently, this project is closed to any external contributions. Any pull request made against this project from external sources will likely be closed. If you would like to make changes to this project, please fork this project.**

# Guide

**This "Help" section can be deleted before submitting this pull request.**

*Update the name of this pull request to reflect the following shape:*

```
{type}/{scope?}/{message}
```

* **type** - A [conventional commit type](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum) **REQUIRED**
* **scope** - The kabab-case scope of the changes in this request
* **message** - A short, kebab-case statement describing the changes **REQUIRED**

*Provide a general summary of the scope of the changes in this pull request.*

* [Readme](https://github.com/momentum-design/momentum-react-v2/blob/master/README.md)
* [Getting Started Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/GETTING_STARTED.md)
* [Contributing Guide](https://github.com/momentum-design/momentum-react-v2/blob/master/CONTRIBUTING.md)

# Description

Using icon names as a type from the momentum-design/icons repo for autocomplete, and selection of types

# Links

*Links to relevent resources.*

### CODE REVIEW

File: src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap [hunk 1]
```diff
@@ -748,7 +748,7 @@ exports[`<AvatarMeetingsListItem /> snapshot should match snapshot with displayM
                       >
                         <Icon
                           fillColor={false}
-                          name="audio-microphone-on-green-colored"
+                          name="microphone-on"
```
Comment (stanjiawang) [inline src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap:751]: Probably need to add ```color``` or style here to add the ```green``` color.

File: src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap [hunk 2]
```diff
@@ -763,7 +763,7 @@ exports[`<AvatarMeetingsListItem /> snapshot should match snapshot with displayM
                               className=""
                               data-autoscale={false}
                               data-scale={16}
-                              data-test="audio-microphone-on-green-colored"
+                              data-test="microphone-on"
```
Comment (stanjiawang) [inline src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap:766]: Probably need to add ```color``` or style here to add the ```green``` color.

File: src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap (hunk 1 above)
Comment (chrisadubois) [reply src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap:751]: not sure it's needed for the unit test - it's up to the consumer

Comment (stanjiawang) [reply src/components/AvatarMeetingsListItem/AvatarMeetingsListItem.unit.test.tsx.snap:751]: hmmm, that makes sense!

**Review by stanjiawang — APPROVED** [review]

**Pull request merged**